Sources/MuxUploadSDK/Public API/MuxUpload.swift

```diff
@@ -105,7 +105,12 @@ public final class MuxUpload {
     /**
      True if this upload is currently in progress and not paused
      */
-    var inProgress: Bool { get { fileWorker != nil } }
+    public var inProgress: Bool { get { fileWorker != nil && !complete } }
```

Comment:
I think this API can also get folded into the changes to `MuxUpload` state that expose an enumerated state that I'm working on right now, what do you think?

Since the example app needs this info we can leave it exposed as-is for now, I'll rebase my branch on this one and make the corresponding adjustments to this API there (and the example code that consumes it too).

Reply:
Also think SDK clients may wish to subscribe for updates to know when `inProgress` and `complete` will change, so as to trigger UI updates or some other effect.

A simple way to enable that is to use the `@Published` property wrapper. (I'm planning on adding something along these lines to the `MuxUpload` state.)

Reply:
I strongly disagree about adding SwiftUI constructs to the SDK surface. Delegates may require boilerplate, but this is a place where it would be trivial to stay unopinionated regarding SwifyUI vs UIKit

Reply:
> I think this API can also get folded into the changes to MuxUpload state that expose an enumerated state that I'm working on right now, what do you think?

Two things:
1) The SDK already has async notifications for completion, by delegate
2) The SDK needs *both* async and synchronous ways to access the state of the operation. These should stay because both use cases are required in basically any app (especially one that uses declarative UIs.

If we preserve both methods of being notified about state, then we can easily support swiftUI without making our API surface this opinionated

Reply:
> I strongly disagree about adding SwiftUI constructs to the SDK surface. Delegates may require boilerplate, but this is a place where it would be trivial to stay unopinionated regarding SwiftUI vs UIKit

I agree about staying unopinionated re SwiftUI vs UIKit. Worth noting that `Published` is [a part of Combine](https://developer.apple.com/documentation/combine/published) and because it hews closely to the venerable Objective-C based key-value observing mechanism that UIKit heavily uses, it can slot nicely with existing UIKit-based code.

Also a (very forward-looking) aside: we ought to consider extending the SDK surface API to make use of [language-level observability constructs](https://github.com/apple/swift-evolution/blob/main/proposals/0395-observability.md) when these are officially added to the language. That design is in broad strokes similar to `Published`, allowing for a relatively simple transition path. The big benefit there is this API can then evolve alongside Swift.

Reply:
> > I think this API can also get folded into the changes to MuxUpload state that expose an enumerated state that I'm working on right now, what do you think?
> 
> Two things:
> 
> 1. The SDK already has async notifications for completion, by delegate. The properties are returning cached versions of the asynchronously-updated values
> 2. The SDK needs _both_ async and synchronous ways to access the state of the operation. These should stay because both use cases are required in basically any app (especially one that uses a declarative UI like swiftUI).
> 
> If we preserve both methods of knowing-about state, then we can easily support swiftUI without making our API surface this opinionated

Ah to be clear the original comment was talking about converting `inProgress` and `complete` to become cases in a `MuxUpload.State` enum. (Details like the exact API spelling are to be nailed down in the input standardization PR, so this was a broader question about approach.)

IMO notifications / state access, whatever they shake out to be, should be consistent across the SDK surface.

Reply:
Ohh!

Yes, I am in favor of publishing a state enum and wrapping data about the upload in it

I'm not opinionated on notification method, if `@Published` doesn't tie us to anything specifically then it seems like a good choice

Reply:
Sounds good, resolving since there's no follow-up needed here. I'll call changes to these out in my PR, and we can then iterate on the specifics.